src/nvim/getchar.c

```diff
@@ -1090,9 +1093,11 @@ static void gotchars(char_u *chars, int len)
   char_u buf[2];
   int todo = len;
 
-  /* remember how many chars were last recorded */
-  if (Recording)
-    last_recorded_len += len;
+  // remember how many chars were last recorded
+  if (Recording) {
+    assert(len >= 0);
```

Comment:
we can make `size_t len` instead and avoid the assert. (downcounting loop using postfix is still safe)